test/integration/tasks.test.js

```diff
@@ -264,6 +264,29 @@ describe("Tasks", function () {
   });
 
   describe("GET /tasks/:id/details", function () {
+    it("Should get tasks filtered by search term", function (done) {
+      const searchTerm = "search";
+      chai
+        .request(app)
+        .get(`/tasks?q=${encodeURIComponent(searchTerm)}`)
```

Comment:
here everything you type in q will become search term. instead we should use `q=searchTerm:golang` so that later we can modify it to `q=searchTerm:golang+assignee:vinit` and so on

Reply:
done